packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -47,9 +49,97 @@ export class TypeSpecCodeActionProvider implements vscode.CodeActionProvider {
         );
       }
     });
+
+    actions.push(...(await this.createInstallPackageByNpm(context, actions)));
+
     return actions;
   }
 
+  private async createInstallPackageByNpm(
+    context: vscode.CodeActionContext,
+    actions: vscode.CodeAction[],
+  ): Promise<vscode.CodeAction[]> {
+    // When the corresponding node dependency package is not installed,
+    // consider generating a quick fix to install via npm command
+    const diagnostics = context.diagnostics.filter(
+      (diagnostic) =>
+        diagnostic.source === "TypeSpec" &&
+        diagnostic.code &&
+        diagnostic.code === "import-not-found",
+    );
+    if (diagnostics.length === 0) {
+      return [];
+    }
+
+    for (const diagnostic of diagnostics) {
+      if (
+        "data" in diagnostic &&
+        diagnostic.data &&
+        typeof diagnostic.data === "object" &&
+        "file" in diagnostic.data
+      ) {
+        // The message content is `Couldn't resolve import "@typespec/http"`.
+        // The compiler's diagnostics do not provide a specific attribute to obtain the package name,
+        // so a regular expression is used to extract the package name within the double quotes
+        const targetPackage = diagnostic.message.replace(/.*"([^"]+)".*/, "$1");
+        const packageFile = diagnostic.data.file as string;
+        const packageFileUri = vscode.Uri.parse(packageFile);
+        const { packageJsonFolder, packageJson } = await searchAndLoadPackageJson(
+          getDirectoryPath(packageFileUri.fsPath),
+        );
+
+        let packageName: string = "";
+        let projectFolder: string | undefined;
+        if (packageJsonFolder === undefined) {
+          // If the package.json file is not found, install the specified `targetPackage` directly
+          packageName = targetPackage;
+          projectFolder = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
```

Comment:
i dont think it's a good idea to install to the root folder of workspace[0] silently, which may not be what user expect. I would prefer not to provide quick fix in this case. Or provide a way for user to choose but it's likely a bigger stuff user needs to fix if user doesn't have package.json which should be  handled by a quick action.